src/main.ts

```diff
@@ -15,6 +16,11 @@ if ('serviceWorker' in navigator) {
     });
 }
 
+// Redirect to Safe (default) if referrer is empty
+if (!document.referrer) {
```

Comment:
Is this necessary here, when the RpcServer already handles it?

This might break the Accounts Manager for browser extensions again, as they do not submit a referrer, I think. Also, does this not trigger for popups?

Reply:
It does not trigger for popups. But you're right, client timeout will happen anyway. Removed it.